psl/psl-core/src/validate/validation_pipeline/validations/indexes.rs

```diff
@@ -386,20 +386,23 @@ pub(crate) fn opclasses_are_not_allowed_with_other_than_normal_indices(index: In
     }
 }
 
-pub(crate) fn composite_types_are_not_allowed_in_index(index: IndexWalker<'_>, ctx: &mut Context<'_>) {
-    for field in index.fields() {
-        if field.scalar_field_type().as_composite_type().is_some() {
-            let message = format!(
-                "Indexes can only contain scalar attributes. Please remove {:?} from the argument list of the indexes.",
-                field.name()
-            );
-            ctx.push_error(DatamodelError::new_attribute_validation_error(
-                &message,
-                index.attribute_name(),
-                index.ast_attribute().span,
-            ));
-            return;
-        }
+pub(crate) fn composite_type_in_compound_index(index: IndexWalker<'_>, ctx: &mut Context<'_>) {
+    let composite_type = index
+        .fields()
+        .find(|f| f.scalar_field_type().as_composite_type().is_some());
+
+    if index.fields().len() > 1 && composite_type.is_some() {
+        
+        let message = format!(
+            "Prisma does not currently support composite types in compound indices. Please remove {:?} from the index.",
```

Comment:
```suggestion
            "Prisma does not currently support composite types in compound indices when used in conjunction with `prisma generate`. Please remove {:?} from the index.",
```
?

Reply:
That is not a thing we would usually say.
It either works, or it doesn't.
With this new validation, it does not work anywhere and is completely blocked from being used.

But you could add a link to the issue tracking the support if you want.

Reply:
This is what I was trying to discuss [above](https://github.com/prisma/prisma-engines/pull/4401#discussion_r1378340908)

Reply:
(Responded there - think I did not see that comment yesterday.)